src/public/app/containers/Song.js

```diff
@@ -177,119 +171,148 @@ class Song extends React.Component {
       credentials: 'include',
       body: JSON.stringify(track),
     })
-    .then(res => res.json())
-    .then(favs => this.props.handleFavoritesChange(favs))
-    .catch(err => console.log(err));
+      .then(res => res.json())
+      .then(favs => this.props.handleFavoritesChange(favs))
+      .catch(err => console.log(err));
   }
 
   handleFavoritesClick(track) {
     if (this.props.favorites.some(fav => fav.track_id === track.track_id)) {
       this.removeFavorite(track);
-    } else this.addFavorite(track);
+    } else {
+      this.addFavorite(track);
+    }
   }
 
-  addToQueue() { this.props.addTrackToSpotifyQueue(this.props.track); }
+  addToQueue() {
+    this.props.addTrackToSpotifyQueue(this.props.track);
+  }
 
   render() {
-    this.netSize = this.props.size - this.borderWidth;
+    const {
+      auth,
+      currentSong,
+      favorites,
+      handleExpandClick,
+      playlist,
+      ranking,
+      spotifyPlayer,
+      track_preview_url,
+      track,
+    } = this.props;
+  
     // TODO fix this piece of logic
     let icon = 'play';
-    if(this.props.auth) {
-      if (this.props.auth &&
-          this.props.spotifyPlayer.currentTrack &&
-          !this.props.spotifyPlayer.isPaused &&
-          this.props.track.track_id === this.props.spotifyPlayer.currentTrack.track_id) {
+    if (auth) {
+      if (auth && spotifyPlayer.currentTrack && !spotifyPlayer.isPaused && (track.track_id === spotifyPlayer.currentTrack.track_id)) {
         icon = 'pause';
-      } else if (!this.props.auth &&
-          this.props.track_preview_url === this.props.currentSong.track_preview_url) {
+      } else if (!auth && (track_preview_url === currentSong.track_preview_url)) {
```

Comment:
`!auth` check inside of a check for `auth`

Reply:
This if/else statement needs a refactor, I just destructured props and removed all instances of this.props